theseus/optimizer/tests/test_manifold_gaussian.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import copy
+
+import numpy as np
+import pytest  # noqa: F401
+import torch
+
+import theseus as th
+from theseus.optimizer.manifold_gaussian import ManifoldGaussian
+
+
+def random_manifold_gaussian_params():
+    manif_types = [th.Point2, th.Point3, th.SE2, th.SE3, th.SO2, th.SO3]
+
+    n_vars = np.random.randint(1, 5)
+    batch_size = np.random.randint(1, 100)
+    mean = []
+    dof = 0
+    for i in range(n_vars):
+        ix = np.random.randint(len(manif_types))
+        var = manif_types[ix].rand(batch_size)
+        mean.append(var)
+        dof += var.dof()
+
+    if np.random.random() < 0.5:
+        precision = torch.zeros(mean[0].shape[0], dof, dof)
+    else:
+        precision = None
+
+    return mean, precision
+
+
+def test_init():
+    all_ids = []
+    for i in range(100):
```

Comment:
Is probably sufficient for all of these loop to be range(10) so CI isn't slowed down.